components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -373,6 +382,68 @@ public RuleTemplate getRuleTemplate(String templateGroupUUID, String ruleTemplat
+        RuleTemplate outputRuleTemplate = getRuleTemplate(businessRuleFromScratch.getTemplateGroupUUID(),
+                businessRuleFromScratch.getOutputRuleTemplateUUID());
+
+        String scriptwithInputTemplatedElement = inputRuleTemplate.getScript();
+        String scriptwithOutputTemplatedElement = outputRuleTemplate.getScript();
+
+        String runnableIntpuScript = TemplateManagerHelper.replaceRegex(scriptwithInputTemplatedElement,
+                TemplateManagerConstants.TEMPLATED_ELEMENT_NAME_REGEX_PATTERN, businessRuleFromScratch.getProperties()
+                        .getInputData());
+        String runnableOutputScript = TemplateManagerHelper.replaceRegex(scriptwithOutputTemplatedElement,
+                TemplateManagerConstants.TEMPLATED_ELEMENT_NAME_REGEX_PATTERN, businessRuleFromScratch.getProperties()
+                        .getOutputData());
+        Map<String, String> inputScriptGeneratedValue = TemplateManagerHelper.getScriptGeneratedVariables
+                (runnableIntpuScript);
+        Map<String, String> outputScriptGeneratedVariables = TemplateManagerHelper.getScriptGeneratedVariables
+                (runnableOutputScript);
+
+        Collection<Template> templatesToBeUsed = getTemplates(businessRuleFromScratch);
+        Map<String, String> inputPropertiesToMap = businessRuleFromScratch.getProperties().getInputData();
+        Map<String, String> outputPropertiesToMap = businessRuleFromScratch.getProperties().getOutputData();
+
+        inputPropertiesToMap.putAll(inputScriptGeneratedValue);
+        outputPropertiesToMap.putAll(outputScriptGeneratedVariables);
+
+        for (int i = 0; i < templatesToBeUsed.size(); i++) {
+            Template[] templates = (Template[]) templatesToBeUsed.toArray();
+            if (i == 0) {
+                templates[1].getType().equals(TemplateManagerConstants.TEMPLATE_TYPE_SIDDHI_APP);
+                Artifact derivedSiddhiApp = deriveSiddhiApp(templates[i], inputPropertiesToMap);
+                try {
+                    // Put SiddhiApp's name and content to derivedTemplates HashMap
+                    derivedTemplates.put(TemplateManagerHelper.getSiddhiAppName(derivedSiddhiApp), derivedSiddhiApp);
+                } catch (TemplateManagerException e) {
+                    log.error("Error in deriving SiddhiApp", e);
+                }
+            }
+            if (i==1){
+                templates[1].getType().equals(TemplateManagerConstants.TEMPLATE_TYPE_SIDDHI_APP);
+                Artifact derivedSiddhiApp = deriveSiddhiApp(templates[i], outputPropertiesToMap);
+                try {
+                    // Put SiddhiApp's name and content to derivedTemplates HashMap
+                    derivedTemplates.put(TemplateManagerHelper.getSiddhiAppName(derivedSiddhiApp), derivedSiddhiApp);
+                } catch (TemplateManagerException e) {
+                    log.error("Error in deriving SiddhiApp", e);
+                }
+            }
+        }
+
```

Comment:
format the code properly unnecessary new lines

Reply:
Fixed by b14dd18f396f799bd32670842869feb2475e778e